code/modules/mob/living/simple_animal/hostile/wumborian_fugu.dm

```diff
@@ -62,65 +66,80 @@
-		to_chat(F, "<span class='warning'>YOU'RE ALREADY WUMBO!</span>")
+/**
+ * Inflates the fugu, making it invulnerable to attack and buffing its offensive and destructive capabilities.
+ *
+ * If the fugu is capable of going wumbo, it will do so.  This gives it invulnerability to damage, high object damage,
+ * and offensive AI as opposed to its default evasive manuvers.  The fugu will deflate after 10 seconds.
+ */
+/mob/living/simple_animal/hostile/asteroid/fugu/proc/Inflate()
+	if(wumbo)
+		to_chat(src, "<span class='warning'>YOU'RE ALREADY WUMBO!</span>")
 		return
-	if(F.inflate_cooldown)
-		to_chat(F, "<span class='warning'>You need time to gather your strength!</span>")
+	if(inflate_cooldown)
+		to_chat(src, "<span class='warning'>You need time to gather your strength!</span>")
 		return
-	if(F.buffed)
-		to_chat(F, "<span class='warning'>Something is interfering with your growth!</span>")
+	if(buffed)
+		to_chat(src, "<span class='warning'>Something is interfering with your growth!</span>")
 		return
-	F.wumbo = 1
-	F.icon_state = "Fugu1"
-	F.obj_damage = 60
-	F.melee_damage_lower = 15
-	F.melee_damage_upper = 20
-	F.harm_intent_damage = 0
-	F.throw_message = "is absorbed by the girth of the"
-	F.retreat_distance = null
-	F.minimum_distance = 1
-	F.move_to_delay = 6
-	F.environment_smash = ENVIRONMENT_SMASH_WALLS
-	F.mob_size = MOB_SIZE_LARGE
-	F.speed = 1
-	addtimer(CALLBACK(F, /mob/living/simple_animal/hostile/asteroid/fugu/proc/Deflate), 100)
+	wumbo = TRUE
+	icon_state = "Fugu1"
+	obj_damage = 60
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	harm_intent_damage = 0
+	throw_message = "is absorbed by the girth of the"
+	retreat_distance = null
+	minimum_distance = 1
+	move_to_delay = 6
+	environment_smash = ENVIRONMENT_SMASH_WALLS
+	mob_size = MOB_SIZE_LARGE
+	set_varspeed(1)
+	addtimer(CALLBACK(src, /mob/living/simple_animal/hostile/asteroid/fugu/proc/Deflate), inflate_time)
 
+/**
+ * Deflates the fugu, resetting it to default state.
+ *
+ * Deflating the fugu undoes all the buffs inflating gives it, setting it back to normal.
+ * This proc also sets the cooldown for the ability, which is by default 4 seconds.
+ */
 /mob/living/simple_animal/hostile/asteroid/fugu/proc/Deflate()
-	if(wumbo)
-		walk(src, 0)
-		wumbo = 0
-		icon_state = "Fugu0"
-		obj_damage = 0
-		melee_damage_lower = 0
-		melee_damage_upper = 0
-		harm_intent_damage = 5
-		throw_message = "is avoided by the"
-		retreat_distance = 9
-		minimum_distance = 9
-		move_to_delay = 2
-		inflate_cooldown = 4
-		environment_smash = ENVIRONMENT_SMASH_NONE
-		mob_size = MOB_SIZE_SMALL
-		speed = 0
+	if(!wumbo)
+		return
+	walk(src, 0)
+	wumbo = 0
+	icon_state = "Fugu0"
+	obj_damage = 0
+	melee_damage_lower = 0
+	melee_damage_upper = 0
+	harm_intent_damage = 5
+	throw_message = "is avoided by the"
+	retreat_distance = 9
+	minimum_distance = 9
+	move_to_delay = 2
+	inflate_cooldown = reinflation_time
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	mob_size = MOB_SIZE_SMALL
```

Comment:
Could we not use `obj_damage = initial(obj_damage)` here?

Reply:
Honestly could use initial for most of these values now I think of it.

Reply:
Yeah sorry, that's what I meant.